core/src/main/java/org/apache/druid/data/input/impl/CloudObjectInputSource.java

```diff
@@ -52,20 +56,30 @@ public CloudObjectInputSource(
       @Nullable List<CloudObjectLocation> objects
   )
   {
+    this.scheme = scheme;
```

Comment:
Nit: it maybe simpler to just call the other constructor from here.

Reply:
@kfaraz I think that's too big of a change especially considering that `FilenameUtils.wildcardMatch` does not document the `null` behavior.

Also, I don't see why `FilenameUtils` cannot be used directly since other `commons` library are already used directly. Creating a method inside `FileUtils` would just add an unnecessary indirection.

Reply:
Sure, @didip , sounds good.